.github/workflows/build-bsp.yml

```diff
@@ -45,6 +45,12 @@ jobs:
         cd boards/${{ matrix.bsp }}
         $(${build_invocation})
 
+    - name: Test Build HAL
```

Comment:
Can you move this into its own job? Where it currently is, it will run for every BSP (~60 times), which is a lot of redundant overhead.

Reply:
Yeah I just realized... I need to read the docs for github actions, nevver used them before

Reply:
Yeah I just realized... I need to read the docs for github actions, never used them before

Reply:
Yeah I just realized... I need to read the docs for github actions, never used them before

Reply:
If you just duplicate + rename the `build` stanza, remove the steps after `Install Rust (thumbv7em)`, remove the `needs` and `strategy` stanzas, and what your step at the end it should work

Oh and change `${{ matrix.toolchain }}` to `stable`